testutils/src/main/java/com/microsoft/identity/internal/testutils/CacheUtils.java

```diff
@@ -0,0 +1,159 @@
+     *
+     * @param SharedPrefName Name of the shared preference where token has been stored.
+     * @param predicate      Generic functional interface representing function that returns true
+     *                       or false depending on taken type.
+     * @param editor         Functional interface to have any number of token editing method.
+     */
+    public void editAllTokenInCache(@NonNull final String SharedPrefName, Predicate<String> predicate, Function<String, String> editor) {
+        SharedPreferences sharedPref = getSharedPreferences(SharedPrefName);
+        SharedPreferences.Editor prefEditor = sharedPref.edit();
+        Map<String, ?> cacheEntries = sharedPref.getAll();
+
+        //get all the key from the cache entry, verify and edit it.
+        for (Map.Entry<String, ?> cacheEntry : cacheEntries.entrySet()) {
+            String keyToEdit = cacheEntry.getKey();
+            if (predicate.test(keyToEdit)) {
+                Object cacheValue = cacheEntries.get(keyToEdit);
+                if (cacheValue instanceof String) {
+                    Credential credential = CACHE_KEY_VALUE_DELEGATE.fromCacheValue((String) cacheValue, Credential.class);
+                    credential.setSecret(editor.apply(credential.getSecret()));
+                    prefEditor.putString(keyToEdit, CACHE_KEY_VALUE_DELEGATE.generateCacheValue(credential));
+                    prefEditor.apply();
+                }
+            }
+        }
+    }
+
+    /**
+     * utility function to edit Access Token.
+     *
+     * @param SharedPrefName Name of the shared preference.
+     * @param editor         Functional interface for token editing method.
+     */
+    public void editAllAccessTokenInCache(@NonNull final String SharedPrefName, Function<String, String> editor) {
+        Predicate<String> predicate = new Predicate<String>() {
+            @Override
+            public boolean test(String cacheKey) {
+                return isAccessToken(cacheKey);
+            }
+        };
+
+        editAllTokenInCache(SharedPrefName, predicate, editor);
+    }
+
+    /**
+     * utility function to edit Refresh Token.
+     *
+     * @param SharedPrefName Name of the shared preference.
+     * @param editor         Functional interface for token editing method.
+     */
+    public void editAllRefreshTokenInCache(@NonNull final String SharedPrefName, Function<String, String> editor) {
+        Predicate<String> predicate = new Predicate<String>() {
+            @Override
+            public boolean test(String cacheKey) {
+                return isRefreshToken(cacheKey);
+            }
+        };
+
+        editAllTokenInCache(SharedPrefName, predicate, editor);
+    }
+
+    private static SharedPreferences getSharedPreferences(final String sharedPrefName) {
+        final Context context = ApplicationProvider.getApplicationContext();
+        return context.getSharedPreferences(sharedPrefName, Context.MODE_PRIVATE);
+    }
```

Comment:
Looks to be the same as this: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java#L56

Is there a need to duplicate it?

Reply:
Methods are private in TestUtils.java file and this file never used therefore I do not understand the purpose this file.

Reply:
It is indeed used. It is currently being used in tests in MSAL and Broker. See this MSAL Test for instance: https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/AcquireTokenNetworkTest.java#L167

Also you can make them public if you want.

Reply:
Changes has been made accordingly.